client/lib/rpc.rs

```diff
@@ -0,0 +1,279 @@
+impl RpcCall {
+    /// Creates a new RPC instance.
+    ///
+    /// `rpc_id` is used for RPC-ID as required by the JSON-RPC specification, and is returned by
+    /// the node in the corresponding response.
+    ///
+    /// `node_address` identifies the network address of the target node's HTTP server, e.g.
+    /// `"http://127.0.0.1:7777"`.
+    ///
+    /// When `verbose` is `true`, the request will be printed to `stdout`.
+    pub fn new(rpc_id: u32, node_address: String, verbose: bool) -> Self {
+        Self {
+            rpc_id,
+            node_address,
+            verbose,
+        }
+    }
+
+    /// Gets a `Deploy` from the node.
+    pub fn get_deploy(self, deploy_hash: DeployHash) -> Result<JsonRpc> {
+        let params = GetDeployParams { deploy_hash };
+        GetDeploy::request_with_map_params(self, params)
+    }
+
+    /// Queries the node for an item at `key`, given a `path` and a `state_root_hash`.
+    /// - `state_root_hash` - Hash representing a pointer to the current state from which a value
+    ///   can be extracted. See `get_state_root_hash`.
+    /// - `key` - Key from which to get a value. See (Key)[casper_types::Key].
+    /// - `path` - Path components starting with the `key`.
+    pub fn get_item(self, state_root_hash: Digest, key: Key, path: Vec<String>) -> Result<JsonRpc> {
+        let params = GetItemParams {
+            state_root_hash,
+            key: key.to_formatted_string(),
+            path,
+        };
+        GetItem::request_with_map_params(self, params)
+    }
+
+    /// Queries the node for the most recent state root hash or as of a given block hash if
+    /// `maybe_block_hash` is provided.
+    pub fn get_state_root_hash(self, maybe_block_hash: Option<BlockHash>) -> Result<JsonRpc> {
+        match maybe_block_hash {
+            Some(block_hash) => {
+                let params = GetStateRootHashParams { block_hash };
+                GetStateRootHash::request_with_map_params(self, params)
+            }
+            None => GetStateRootHash::request(self),
+        }
+    }
+
+    /// Gets the balance from a purse.
+    /// - `state_root_hash` - Hash representing a pointer to the current state from which a balance
+    ///   can be extracted. See `get_state_root_hash`.
+    /// - `purse_uref` - Represents the purse from which the balance will be obtained. Can be
+    ///   obtained with `get_item`.
+    pub fn get_balance(self, state_root_hash: Digest, purse_uref: URef) -> Result<JsonRpc> {
+        let params = GetBalanceParams {
+            state_root_hash,
+            purse_uref: purse_uref.to_formatted_string(),
+        };
+        GetBalance::request_with_map_params(self, params)
+    }
+
+    /// Transfers an amount between purses.
+    /// - `amount` - Specifies the amount to be transferred.
+    /// - `source_purse` - Source purse in the sender's account that this amount will be drawn from,
+    ///   if it is not specified, it will defaul to their main purse.
```

Comment:
```suggestion
    /// - `source_purse` - Source purse in the sender's account that this amount will be drawn from.
    ///   If it is `None`, it will default to the account's main purse.
```